clients/src/main/java/org/apache/kafka/common/security/auth/KafkaPrincipal.java

```diff
@@ -53,7 +53,12 @@ public KafkaPrincipal(String principalType, String name) {
     }
 
     public KafkaPrincipal(String principalType, String name, boolean tokenAuthenticated) {
-        this.principalType = requireNonNull(principalType, "Principal type cannot be null");
+        //Sharing USER_TYPE reference.
+        if (principalType.equals(USER_TYPE)) {
```

Comment:
This moved the equals check to the initialization of the object. instead of the check itself.